frame/contracts/src/lib.rs

```diff
@@ -1210,6 +1268,21 @@ impl<T: Config> Invokable<T> for InstantiateInput<T> {
 	}
 }
 
+macro_rules! ensure_no_migration_in_progress {
```

Comment:
we initially used an .into() from an Error but Alex was worried that it would be a foot gun to wrongly convert other errors into ContractResult.  Creating a new method could lead to similar issue, one could use it in the wrong place and ends up with all these ::zero values in place where you might actually have some gas / deposit consumed.